components/brave_wallet/browser/simulation_request_helper.cc

```diff
@@ -0,0 +1,189 @@
+    mojom::TransactionInfoPtr tx_info) {
+  base::Value::Dict tx_object;
+  tx_object.Set("from", tx_info->from_address);
+
+  if (tx_info->tx_data_union->is_eth_tx_data_1559()) {
+    const auto& tx_data = tx_info->tx_data_union->get_eth_tx_data_1559();
+
+    // TODO(onyb): Convert hex to wei string
+    //
+    // Ethereum JSON-RPC API requires value to be in wei formatted as a string.
+    // Using a 0x-prefixed hex string also works, but is neither documented in
+    // the ETH JSON-RPC spec nor in Blowfish docs. Hex strings are widely used
+    // in our codebase, so we'll use them here for consistency. This comment
+    // apples to all other places where we use hex strings.
+    //
+    // Here's an unoptimised implementation for future reference:
+    //
+    // absl::optional<std::string> HexToWei(const std::string& value) {
+    //   uint256_t uint256_value;
+    //   if (!HexValueToUint256(value, &uint256_value)) {
+    //     return absl::nullopt;
+    //   }
+    //
+    //   std::string result;
+    //   while (uint256_value > 0) {
+    //     uint64_t digit = uint256_value % 10;
+    //     result = std::to_string(digit) + result;
+    //     uint256_value /= 10;
+    //   }
+    //
+    //   return result;
+    // }
+    tx_object.Set("value", tx_data->base_data->value);
+    tx_object.Set("to", tx_data->base_data->to);
+    tx_object.Set("data", ToHex(tx_data->base_data->data));
+  } else if (tx_info->tx_data_union->is_eth_tx_data()) {
+    const auto& tx_data = tx_info->tx_data_union->get_eth_tx_data();
+
+    tx_object.Set("value", tx_data->value);
+    tx_object.Set("to", tx_data->to);
+    tx_object.Set("data", ToHex(tx_data->data));
+  } else {
+    return absl::nullopt;
+  }
+
+  base::Value::Dict params;
+  params.Set("txObject", std::move(tx_object));
+  params.Set("metadata", GetMetadata(std::move(tx_info->origin_info)));
+  params.Set("userAccount", tx_info->from_address);
+
+  return GetJSON(base::Value(std::move(params)));
+}
+}  // namespace evm
+
+namespace solana {
+absl::optional<std::string> GetBase64TransactionFromTxDataUnion(
+    mojom::TxDataUnionPtr tx_data_union) {
+  if (!tx_data_union->is_solana_tx_data()) {
+    return absl::nullopt;
+  }
+
+  auto tx = SolanaTransaction::FromSolanaTxData(
+      std::move(tx_data_union->get_solana_tx_data()));
+
+  auto message_signers_pair = tx->GetSerializedMessage();
+  if (!message_signers_pair) {
+    return absl::nullopt;
+  }
+
+  auto& message_bytes = message_signers_pair->first;
+  auto& signers = message_signers_pair->second;
+  if (signers.size() > UINT8_MAX) {
+    return absl::nullopt;
+  }
+
+  // transaction_bytes is a compact-array of signatures, followed by the
+  // message bytes. Since the transactions are not signed yet, we use 64
+  // empty bytes for each signature.
+  std::vector<uint8_t> transaction_bytes;
+  CompactU16Encode(signers.size(), &transaction_bytes);
+  transaction_bytes.insert(transaction_bytes.end(), 64 * signers.size(), 0);
```

Comment:
64 to named constant please

Reply:
e2381cf3e1f72278ef34bfdf2eddb32ecc222951